src/DotNetLightning.Core/Channel/Channel.fs

```diff
@@ -107,6 +107,19 @@ module Channel =
                 failwith "TODO"
             }
 
+    let makeChannelReestablish (data: Data.IHasCommitments): Result<ChannelEvent list, ChannelError> =
+        let commitmentSeed = data.Commitments.LocalParams.ChannelPubKeys.CommitmentSeed
+        let ourChannelReestablish =
+            {
+                ChannelId = data.ChannelId
+                NextLocalCommitmentNumber = 1UL
+                NextRemoteCommitmentNumber = 0UL
+                DataLossProtect = OptionalField.Some({
+                                      YourLastPerCommitmentSecret = PaymentPreimage([|for _ in 0..31 -> 0uy|])
```

Comment:
what is 31?

Reply:
32 minus one. PaymentPreimage will check the length. The same idiom is used elsewhere, see https://github.com/joemphilips/DotNetLightning/blob/75a19767fe7c4bf23a7bf878906a066867f7fe03/tests/DotNetLightning.Core.Tests/Serialization.fs#L604

Reply:
and what is 32? I mean is this a magic number?

Reply:
It is defined in bolt-2 that `payment_preimage` must be 32 bytes.
In general, when we have some kind of secret-and-commitment scheme in bitcoin, usually 32 bytes secret is used (even though technically it can be longer)
This is because 32 bytes secret is the shortest length to assure the enough entropy for its hash.

Reply:
But I don't understand why it is ok here to use non-random secret. Shouldn't be some kind of random bytes?

Reply:
shouldn't the consumer of DotNetLightning be the one in charge to provide the randomness source instead of DNL itself?

Reply:
Yes, that is the purpose for `IKesRepository` .

Reply:
@knocte If no commitment has been revoked, no secret is available. The channel can be reestablished before a commitment has been revoked. A secret is revoked as part of the `revoke_and_ack` message. If we have not received that message, no secret is available. Furthermore, that code is existing, and this commit is concerned with moving it out into its own function so that we can restore channels in the Normal state also, which is necessary since Geewallet now saves channels so that they have Normal state on disk. I will replace this code with something better once the time is right. But there are bound to be other bugs concerning adding and removing commitments, so I think it is premature to worry about if this piece of code is general enough.

Reply:
I was just nitpicking about magic numbers, which I fixed in https://github.com/joemphilips/DotNetLightning/pull/45 (WIP)

Reply:
Will at least this PR is better than before. So I suppose I should just merge this and think about the plausibility of using  `[|for _ in 0..31 -> 0uy|]` as a preimage later.